### PR TITLE
Fix target_link_libraries calls w/ alias target

### PR DESCRIPTION
**Pull request by nmnobre**
Hey @ax3l, @atmyers, @WeiqunZhang,

I think the guided tutorials are no longer building?
Feel free to edit with a better/shorter fix. :)

Cheers,
-Nuno

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
It seems to build for me. What issues do you see?
```
$ amrex-tutorials/GuidedTutorials/HelloWorld/
$ mkdir build
$ cd build
$ cmake ..
$ mpiexec -n 4 ./HelloWorld
Initializing AMReX (24.12-14-gb3f67385e62f)...
MPI initialized with 4 MPI processes
MPI initialized with thread support level 0
AMReX (24.12-14-gb3f67385e62f) initialized
Hello world from AMReX version 24.12-14-gb3f67385e62f
AMReX (24.12-14-gb3f67385e62f) finalized
```

**Comment by WeiqunZhang** [comment]
I am using
```
commit 51fabc1563c3949a364e1a091d2632d4974662a8 (HEAD -> main, origin/main, origin/HEAD)
Merge: ffbfc5a d261a37
Author: Andy Nonaka <AJNonaka@lbl.gov>
Date:   Thu Dec 19 13:01:35 2024 -0800

    Merge pull request #148 from ajnonaka/main
    
    Add 2d to basic fft tutorial
```

**Comment by nmnobre** [comment]
I can't tell from your output for certain, but make sure you're using an existing AMReX installation with `-DAMReX_ROOT=...`, in which case you should see: 

```
CMake Error at CMakeLists.txt:70 (target_link_libraries):
  target_link_libraries can not be used on an ALIAS target.
```

**Comment by WeiqunZhang** [comment]
I see. Let me try again later today.

**Comment by nmnobre** [comment]
Rebased to remove the updates to the CI workflows to install {cu,roc}SPARSE since @ajnonaka leapfrogged me. :)

**Review by ax3l — APPROVED** [review]

File: GuidedTutorials/HeatEquation_Simple/CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +63,8 @@ else()
 
   # Add mpi
   find_package(MPI REQUIRED)
```
[the remark this answers is not in the document]
Comment (nmnobre) [reply GuidedTutorials/HeatEquation_Simple/CMakeLists.txt]: > AMReX has to be built with MPI

Should we check for that? Or because we don't necessarily need MPI, do we just go with whatever the AMReX pkg we find already has (or has not)?

[the remark this answers is not in the document]
Comment (ax3l) [reply GuidedTutorials/HeatEquation_Simple/CMakeLists.txt]: Correct. I pushed a fix :)
